Add CODE_OF_COLLAB.md to Clarify Collaboration and Research Integrity Expectations

yunks128 opened the pull request:
## Purpose  
This PR introduces an `EXPECTATIONS.md` file under `docs/guides/governance/contributions/code-of-conduct/` to address the collaborative and scientific expectations associated with contributing to this project.

## Proposed Changes  
- [ADD] `EXPECTATIONS.md` outlining:
  - Our metrics of success (scientific impact, peer-reviewed publication, research funding)
  - Our open science commitments and collaboration principles
  - Expected conduct and communication norms
  - Research integrity aligned with the European Code of Conduct

## Issues  
- Responds to and resolves [[Issue #173](https://github.com/NASA-AMMOS/slim/issues/173)]

## Testing  
- Rendered and verified Markdown formatting locally  
- Reviewed content for clarity, alignment with open science principles, and scientific collaboration standards

riverma commented:
@yunks128 - thanks for proposing this, couple questions:
- Is this the file that @c-h-david has been promoting? If so - let's work with him on wording
- I suggest we refactor the Code of Conduct page to include two artifacts, not just one, to discuss conduct suggestions. Having the `EXPECTATIONS.md` in the folder but not including it or referencing it in the core [README.md file](https://github.com/NASA-AMMOS/slim/blob/main/docs/guides/governance/contributions/code-of-conduct/README.md) in the docs folder means the `EXPECTATIONS.md` file will be hard to get to. When re-factoring, I think you want to propose that readers use the Contributor Covenant OR this new file OR both.

riverma commented on static/data/slim-registry.json:
```diff
@@ -172,6 +172,11 @@
         "uri": "https://www.contributor-covenant.org/version/2/1/code_of_conduct/code_of_conduct.md"
+      },
+      {
+      "name": "SLIM Collaboration Expectations",
```
I think it's better if we don't brand it with SLIM, but rather a name that conveys the original name proposed by @c-h-david et al, something like: https://github.com/c-h-david/rapid2/blob/main/CODE_OF_COLLAB.md ?

c-h-david replied:
Actually, my initial proposed name was indeed `EXPECTATIONS.md`, but upon further conversations with the SLIM team and after doing some reading, I now recognize that this is really a "code of conduct for collaboration". There is a definition in [this paper](https://doi.org/10.1007/s11213-023-09641-3): "[A code of conduct for collaboration] is a set of values, standards, rules, and principles describing expectations for what members of an organization or participants of a process should do".  As a result, I now think of Contributor Covenant as a "Code of Conduct for Behavior" and [`CODE_OF_COLLAB`](https://github.com/c-h-david/rapid2/blob/main/CODE_OF_COLLAB.md) as a "Code of Conduct for Scientific Collaboration".

hookhua replied:
[CODE_OF_COLLAB](https://github.com/c-h-david/rapid2/blob/main/CODE_OF_COLLAB.md) as a "Code of Conduct for Scientific Collaboration"
Comparing this to the TOPS [CODE_OF_CONDUCT.md](https://github.com/opensciency/Transform-to-Open-Science-2024/blob/main/CODE_OF_CONDUCT.md) and [CONTRIBUTING.md](https://github.com/opensciency/Transform-to-Open-Science-2024/blob/main/CONTRIBUTING.md), the "Code of Conduct for Scientific Collaboration" here actually fills a gap in describing the angle of expected conduct in scientific collaboration. While the https://www.contributor-covenant.org/version/2/1/code_of_conduct/ appears to be more generic. Am wondering if the RAPID2's [CODE_OF_COLLAB.md](https://github.com/c-h-david/rapid2/blob/main/CODE_OF_COLLAB.md) could actually be a useful contribution back to TOPS since that is more focused on the open science process of collaboration?

hookhua replied:
In https://github.com/c-h-david/rapid2/blob//CODE_OF_COLLAB.md#our-metrics-of-success
it currently shows:

```
As a project developed by scientific researchers, our primary metrics of success are:
- Achieving scientific and societal impact.
- Publishing articles in peer-reviewed journals.
- Gathering funding to support ongoing research and development.
```

Could another merit of success be the more basic, but useful:

- Collaboration in the use of a shared open source scientific software.

A bonus if that scientific software collaboration led to PRs community contributions.

c-h-david replied:
I like the idea but I'm afraid working with the community on PRs doesn't do much at the end of the year for a scientist's performance evaluation. However, the kind of dedicated collaborative work that leads to a PR usually comes with a scientific manuscript submitted to a journal. And that's really the second bullet. @riverma and I were actually brainstorming yesterday about trying to map the `GOVERNANCE.md` table with roles in publications. That could help tie things together.

c-h-david replied:
Also, @hookhua, we had an issue open with NASA Open Science (formerly NASA TOPS) that is referenced in [CODE_OF_COLLAB.md](https://github.com/c-h-david/rapid2/blob/main/CODE_OF_COLLAB.md) but it appears that the TOPS repo was retired. I'm not sure - what if anything - replaced it.

yunks128 commented:
@riverma Thanks for the helpful review and suggestions! I've updated the Code of Conduct README to include the new artifact, *Code of Conduct for Collaboration*, and also updated the SLIM registry. Let me know what you think!

riverma commented on static/data/slim-registry.json:
```diff
@@ -172,6 +172,11 @@
+      {
+      "name": "Code of Conduct for Collaboration",
+      "type": "text/md",
+      "uri": "https://raw.githubusercontent.com/NASA-AMMOS/slim/main/docs/guides/governance/contributions/code-of-conduct/CODE_OF_COLLAB.md"
```
@yunks128 - can you move this file to `https://raw.githubusercontent.com/NASA-AMMOS/slim/main/static/assets/governance/code-of-conduct`? That sub-folder needs to be created. Putting files here helps docusaurus render contents for files with extensions like `MD` or `YAML` better (prevents errors).

riverma commented on docs/guides/governance/contributions/code-of-conduct/README.md:
```diff
@@ -46,6 +46,19 @@ Access the standard Contributor Covenant template for a robust Code of Conduct p
+We recommend all contributors read both documents:
+
+- Use the [**Contributor Covenant**](https://www.contributor-covenant.org/version/2/1/code_of_conduct/code_of_conduct.md) for general behavior and inclusive participation in open-source settings.
+- Refer to [**CODE_OF_COLLAB.md**](./CODE_OF_COLLAB.md) to understand our **collaboration expectations** in the context of research, publication, and funding goals.
```
@yunks128 - can you add this file as a downloadable file within the Quick Start section as well? Similar to how we're pointing to the existing code of conduct recommendation.

riverma commented on docs/guides/governance/contributions/code-of-conduct/README.md:
```diff
@@ -46,6 +50,19 @@ Access the standard Contributor Covenant template for a robust Code of Conduct p
+
+In addition to the standard Contributor Covenant, this project includes a supplementary document: [`CODE_OF_COLLAB.md`](https://raw.githubusercontent.com/NASA-AMMOS/slim/main/static/assets/governance/code-of-conduct/CODE_OF_COLLAB.md), which outlines expectations specific to scientific research collaboration.
+
+We recommend all contributors read both documents:
```
Consider changing the wording here to be more specific about recommending the code of conduct for most, if not all soft projects, and the code of collab specifically for scientific software.

riverma commented on docs/guides/governance/contributions/code-of-conduct/README.md:
```diff
@@ -46,6 +50,19 @@ Access the standard Contributor Covenant template for a robust Code of Conduct p
+- Use the [**Contributor Covenant**](https://www.contributor-covenant.org/version/2/1/code_of_conduct/code_of_conduct.md) for general behavior and inclusive participation in open-source settings.
+- Refer to [**CODE_OF_COLLAB.md**]([./CODE_OF_COLLAB.md](https://raw.githubusercontent.com/NASA-AMMOS/slim/main/static/assets/governance/code-of-conduct/CODE_OF_COLLAB.md)) to understand our **collaboration expectations** in the context of research, publication, and funding goals.
+
+You may adopt one or both depending on your community and objectives.
```
Same here – or combined this with the above recommendations.

riverma commented on static/assets/governance/code-of-conduct/CODE_OF_COLLAB.md:
```diff
@@ -0,0 +1,80 @@
+
+3. **Contribute to Mutual Success**  
+   Our open-source work is designed to benefit both you and the scientific community. If you find our software valuable:
+   - **Cite our work** when using it in research.
```
It might be a good idea to link to a citation document here. Should we include, as part of this guide, a template for a citation.MD? If not here might be a good idea to put it someplace else. Get a provides a couple of examples. Perhaps we can include this as a template.

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files

riverma commented on CITATION.cff:
```diff
@@ -0,0 +1,88 @@
+type: software
+title: "Software Lifecycle Improvement & Modernization (SLIM)"
+abstract: "A shared community resource for discussing, iterating and referencing a common set of best practices in software development standards for NASA multi-mission space and ground software."
+authors:
```
The authors list is always evolving and there could be some folks contributing more in words than code. Can we remove the authors list entirely and just keep the citation to the SLIM project itself?

riverma reviewed (approved):
LGTM! Thanks @yunks128
